src/Swarm/Game/Step.hs

```diff
@@ -2048,30 +2055,33 @@ execConst c vs s k = do
        in throwError . Fatal $ msg <> badConstMsg
  where
   doDrill d = do
-    inv <- use robotInventory
     ins <- use equippedDevices
 
     let equippedDrills = extantElemsWithCapability CDrill ins
         -- Heuristic: choose the drill with the more elaborate name.
         -- E.g. "metal drill" vs. "drill"
         preferredDrill = listToMaybe $ sortOn (Down . T.length . (^. entityName)) equippedDrills
 
-    drill <- preferredDrill `isJustOr` Fatal "Drill is required but not equipped?!"
+    tool <- preferredDrill `isJustOr` Fatal "Drill is required but not equipped?!"
+    applyDevice ins "drill" d tool
 
-    (nextLoc, nextE) <- getDrillTarget "drill" d
+  applyDevice ins verbPhrase d tool = do
+    (nextLoc, nextE) <- getDrillTarget verbPhrase d
```

Comment:
It might be nice to clean up the remaining variable names with `drill` in them and make them more generic, like `getDeviceTarget` instead of `getDrillTarget`.

Reply:
Fixed in #1299